examples/SharedMemory/PhysicsDirect.cpp

```diff
@@ -55,12 +55,17 @@ struct PhysicsDirectInternalData
 
 	char    m_bulletStreamDataServerToClient[SHARED_MEMORY_MAX_STREAM_CHUNK_SIZE];
 	btAlignedObjectArray<double> m_cachedMassMatrix;
+	int m_cachedCameraArraySize;
 	int m_cachedCameraPixelsWidth;
 	int m_cachedCameraPixelsHeight;
+	int m_cachedCameraFeatureLength;
 	btAlignedObjectArray<unsigned char> m_cachedCameraPixelsRGBA;
 	btAlignedObjectArray<float> m_cachedCameraDepthBuffer;
 	btAlignedObjectArray<int> m_cachedSegmentationMask;
-	
+
+	unsigned char *m_ptrCameraArrayPixelsRGB;
```

Comment:
This is a dodgy place - as I'm not sure, if a pointer suffices.  It looks like m_cachedCameraXXX storage/memcpy comes from receiving camera data in blocks through shared memory.  I'm making an assumption that only direct mode is meaningful for a plugin / camera array rendering.  And in that mode sending data in blocks / memcpys are avoidable.  

But the ownership of the memory is not very clear to me - so this place is dodgy.  Seem to work though.

Reply:
I would store a copy of the data in the client (such as PhysicsDirect.cpp), and not a pointer. The plugin also works in SHARED_MEMORY_SERVER mode etc.